pkg/controller/metrics/metrics.go

```diff
@@ -265,13 +285,61 @@ func (mc *Calculator) Start(stopCh <-chan struct{}) error {
 			}
 		}
 
+		mc.calculateSelectorSyncSetMetrics(mcLog)
+
 		elapsed := time.Since(start)
 		mcLog.WithField("elapsed", elapsed).Info("metrics calculation complete")
 	}, mc.Interval, stopCh)
 
 	return nil
 }
 
+func (mc *Calculator) calculateSelectorSyncSetMetrics(mcLog log.FieldLogger) {
+	mcLog.Debug("calculating metrics across all SyncSetInstances")
+	ssis := &hivev1.SyncSetInstanceList{}
+	err := mc.Client.List(context.Background(), ssis)
+	if err != nil {
+		mcLog.WithError(err).Error("error listing all SyncSetInstances")
+		return
+	}
+
+	sssInstancesTotal := map[string]int{}
+	sssInstancesUnappliedTotal := map[string]int{}
+
+	ssInstancesTotal := 0
+	ssInstancesUnappliedTotal := 0
+	for _, ssi := range ssis.Items {
+		if ssi.Spec.SelectorSyncSet != nil {
```

Comment:
This can be more concise. You do not need the code that initializes the map entry to 0.
```
if sss := ssi.Spec.SelectorSyncSet; sss != nil {
  sssInstancesTotal[sss.Name]++
  if ssi.Status.Applied {
    sssInstancesUnappliedTotal[sss.Name]++
  }
}
```

Reply:
Much better, thanks.

Reply:
I see that you made the change in the `else` block but not in the `if` block. Was that an oversight?

Reply:
I just pushed recently, not sure if you saw that version, but the only else I see in this area wasn't doing the same thing. (just simple counters going up, no map)

Reply:
Yes, I was looking at the version before the push. I see that it is updated now.